### ISSUE TITLE
False positives for CVE-2016-3674 for several libraries

### ISSUE THREAD

**Issue by shayke** [dependency-check/DependencyCheck#554]
[CVE-2016-3674](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-3674) is falsely reported for the following libraries:
- https://mvnrepository.com/artifact/org.tinygroup/org.tinygroup.xstream/1.2.0
- https://mvnrepository.com/artifact/ro.pippo/pippo-xstream/0.6.0
- https://mvnrepository.com/artifact/de.thksystems/tkscommons-xstream/0.1.1
- https://mvnrepository.com/artifact/com.h3xstream.retirejs/retirejs-core/1.0.0

There are a lot more but my main question here is why those with only "xstream" in the vendor or product are found?
I understand that some of those libraries has xstream as a dependency but that is not what the project is looking for if I'm not mistaken.

Eventually, I would expect to see only the core xstream library as vulnerable, all the others
are found too since they have xstream in their vendor **OR** product name.
Is it possible to tweak the mechanism further?

**Comment by jeremylong** [comment]
Thanks for pointing this specific example out - I'll take a look to see if
something can be done. This is a class of false positives I don't have a
good solution to (yet). There are some planned enhancements to the version
number matching that might help - however, to really understand the problem
take a look at the following articles:
- How does dependency-check work
  http://jeremylong.github.io/DependencyCheck/general/internals.html
- How to read the report
  http://jeremylong.github.io/DependencyCheck/general/thereport.html

Then to deal with the problem we can currently use suppression filters. In
general, it does not take long to onboard an application (i.e. create a
suppression filter for the false positives). More on creating the
suppression filter can be found here
http://jeremylong.github.io/DependencyCheck/general/suppression.html. I
would generally recommend using a suppression filter based on the GAV, make
it a regex, and just leave the version numbers off of the GAV and CPE
combination that are being suppressed as you can see as the last example in
the suppression file examples.

**Comment by shayke** [comment]
Thanks!
Can you share some info about those future enhancements?
I'm not sure if the versioning is the problem here but rather the product matching, is there an open discussion about future plans somewhere?

**Comment by jeremylong** [comment]
The enhancements are around version number matching. One of the issues is that if a library has a lower version number listed (possibly specification version in the manifest, or some other lower quality version number) it may cause a match. There are other cases where better version number matching can decrease the number of false positives. This might help in a few of the cases related to the issue you raised, but it definitely will not fix them all.

The case you've brought up (which is similar to other open issues) will likely require adding a [suppression filter](http://jeremylong.github.io/DependencyCheck/general/suppression.html) to eliminate the false positives..

**Comment by jeremylong** [comment]
Just added a suppression file to the core engine:

``` xml
    <suppress>
        <notes><![CDATA[
        xstream false positives
        ]]></notes>
        <gav regex="true">^(?!com.thoughtworks).*xstream.*$</gav>
        <cpe>cpe:/a:x-stream:xstream</cpe>
    </suppress>
```

This will be included in the 1.4.4 release.
